internal/server/email/smtp_test.go

```diff
@@ -104,6 +104,7 @@ func successCase(t *testing.T, c net.Conn) {
 	write("354 Ready")
 	for s := read(""); s != "."; s = read("") {
 		switch {
+		case strings.HasPrefix(s, "X-SMTPAPI:"):
```

Comment:
How can we test this change? I guess you can manually run the code against a dev sendgrid account? 

Could we also include some automated test coverage here to at least check that the JSON literal is well formed and parses into the expected structure? That should allow someone to make minor changes to this code without needing to setup a dev sendgrid account.

Reply:
if you change this code you're not going to get around testing it manually against sendgrid, as they're the only ones who can say it's valid/working, valid json or not.

Reply:
Similar to how we use golden files to test the CLI output, I think asserting that the value is what is expected is still valuable. It catches problems where the value has changed accidentally without us noticing.

Right now we could completely remove this header and no tests would fail.